### ISSUE TITLE
Evaluates expressions on conditional resources.

### ISSUE THREAD

**Issue by cyrus-mc** [terraform-linters/tflint#613]
Conditional resources are still passed through linting resulting in possible warnings.

```hcl
resource "aws_route53_record" "main" {
  provider = aws.route53

  count = (local.create_route53_records * var.instance_count)

  zone_id = lookup(var.route53, "zone_id")

  name    = aws_instance.main[count.index].tags["Name"]
  type = "A"
  ttl  = lookup(var.route53, "ttl", 300)

  records = [ aws_instance.main[count.index].private_ip ]
}
```

The above code fails (when count would normally evaluate to 0) do to the lookup on var.route53 not having key "zone_id". This is due to the fact a value for zone_id isn't required as this resource is not being created. 

TFLint v0.14.0
Terraform v0.12.20

**Comment by wata727** [comment]
Sorry, I'm not sure what issue you faced... Could you tell me a little more about the problem?
It would be helpful if you could write the expected behavior and the actual result.

**Comment by wata727** [comment]
Close this due to no activity.

**Comment by jmorris0x0** [comment]
Hi. This issue still exists.

### Version
0.31.0

### Current Behavior
In the resource below, if `var.tgw_id` is set to `null`, then the count of the resource below will be zero.

```
resource "aws_route" "example" {
  count = var.tgw_id == null ? 0 : 1

  route_table_id         = aws_route_table.public[count.index].id
  destination_cidr_block = "10.0.0.0/8"
  transit_gateway_id     = var.tgw_id
}
```


But the resource is linted anyway and an error is returned due to `transit_gateway_id = null`.
```
Error: The routing target is not specified, each aws_route must contain either egress_only_gateway_id, gateway_id, instance_id, nat_gateway_id, network_interface_id, transit_gateway_id, vpc_peering_connection_id or vpc_endpoint_id. (aws_route_not_specified_target)

  on main.tf line 754:
 754: resource "aws_route" "example" {

Reference: https://github.com/terraform-linters/tflint-ruleset-aws/blob/v0.6.0/docs/rules/aws_route_not_specified_target.md
```

### Expected Behavior

The resource should not be linted because it will not exist when `count` is `0`

**Comment by bendrucker** [comment]
The original issue is invalid as the count is not evaluable by TFLint. See #571. 

TFLint does attempt to handle this:

https://github.com/terraform-linters/tflint/blob/10c6c1549ed595cfe97434831a955603a9cb693a/tflint/runner_eval.go#L314-L352

If you think you have a similar problem, open a new issue including both the configuration _and_ the command you ran to invoke TFLint so the output is reproducible.
